### DISCUSSION TITLE
How sensible are rsa_key* ?

### DISCUSSION THREAD

**Discussion by CuriousPelican** [dani-garcia/vaultwarden#2311] [Q&A] [answered]
Hello,

First, I don't know whether this is the right place to ask or not, sorry if that's not the case.

I want to run vaultwarden and I am thinking about how to protect my instance from physical attacks.
So I wanted to know if someone with my private RSA key could sign a JWT and access my vault without having my master password.

Thanks (btw sorry if my English is not perfect, it isn't my mother language)

**Comment by jjlin** [comment, marked as answer]
A valid JWT would only provide access to the encrypted data stored for an account (e.g., via the `/api/sync` response), but the master password would still be needed to decrypt that data.

But if someone can obtain the RSA private key from your server, then they can most likely access the database as well, so there's little point trying to forge JWTs when they can just dump the entire database.

**Reply by CuriousPelican** [reply]
> they can just dump the entire database

For sure but data contained in the db is encrypted using master password right ?

Edit: found the answer on [this](https://github.com/dani-garcia/vaultwarden/discussions/1798) discussion

**Reply by jjlin** [reply]
It's actually encrypted with a randomly-generated encryption key that is itself encrypted with a hashed form of your master password, but basically yes.
